lib/mix/tasks/compile.thrift.ex

```diff
@@ -91,13 +100,19 @@ defmodule Mix.Tasks.Compile.Thrift do
     File.rm(manifest)
   end
 
-  @spec parse(Path.t(), OptionParser.parsed()) :: FileGroup.t()
-  defp parse(file, opts) do
-    Thrift.Parser.parse_file_group!(file, opts)
-  rescue
-    e ->
-      Mix.shell().error("Failed to parse #{file}: #{Exception.message(e)}")
-      nil
+  @spec parse([Path.t()], Thrift.Parser.opts()) :: {[FileGroup.t()], [Diagnostic.t()]}
+  defp parse(files, opts) do
+    files
+    |> Enum.reverse()
```

Comment:
The reverse here is so the output is same order as input? Or is there something deeper?

Reply:
That's correct. It's so we can push the following diagnostics onto the head of the `diagnostics` list and the result remains consistent with the ordering of the input files.